base/base/ReplxxLineReader.cpp

```diff
@@ -179,9 +179,8 @@ ReplxxLineReader::ReplxxLineReader(
 
     auto callback = [&suggest] (const String & context, size_t context_size)
     {
-        if (auto range = suggest.getCompletions(context, context_size))
-            return Replxx::completions_t(range->first, range->second);
-        return Replxx::completions_t();
+        auto words = suggest.getCompletions(context, context_size);
```

Comment:
Probably it's better to return `Replxx::completions_t` directly from `getCompletions()`. This is because `Replxx::completions_t` is a vector of `Replxx::Completion`, and `Replxx::Completion` keeps a string inside. So doing this way you're going to copy all the completion twice - the first time while you're collecting words inside `getCompletions()` and the second time while you're constructing `Replxx::completions_t` here.

Reply:
>Probably it's better to return Replxx::completions_t directly from getCompletions(). 

This will break abstractions a little, but I'm fine with this.

>This is because Replxx::completions_t is a vector of Replxx::Completion, and Replxx::Completion keeps a string inside

Right.